Source/Common/IGListBatchUpdateData.mm

```diff
@@ -29,16 +29,45 @@ static void convertMoveToDeleteAndInsert(NSMutableSet<IGListMoveIndex *> *moves,
 
 @implementation IGListBatchUpdateData
 
+/**
+ Finds duplicate index paths and increments increments the item index so eacy operation is unique. Duplicate delete
+ operations cannot be performed within the same batch update block. However deletes have corresponding data-source
+ changes so the operation still has to occur, otherwise UICollectionView will throw an inconsistency exception. See
+ https://github.com/Instagram/IGListKit/issues/651
+ */
+static NSIndexPath *staggeredIndexPath(NSCountedSet<NSIndexPath *> *dupes, NSIndexPath *path) {
+    const NSInteger count = [dupes countForObject:path];
+    if (count == 0) {
+        [dupes addObject:path];
+        return path;
+    }
+
+    NSIndexPath *nextPath = [NSIndexPath indexPathForItem:path.item + count inSection:path.section];
+    NSIndexPath *staggeredPath = staggeredIndexPath(dupes, nextPath);
+    [dupes addObject:staggeredPath];
+    return staggeredPath;
```

Comment:
this method is a bit confusing to read due to the fact that the return value isn't just the new index path (which happens to be the last one generated, i think) but also the implicit change to `dupes`. i'd prefer this to return a struct that returns the `NSCountedSet` and `NSIndexPath` as elements, even if that set is the same reference as what was passed in.